File: tools/remoteserver/example/examplelibrary.rb
class ExampleRemoteLibrary

  def count_files directory
      return Dir.new(directory).entries.find_all{|f| File.file? f}.length
  end 

  def strings_should_be_equal str1, str2
    puts "Comparing '#{str1}' to '#{str2}'"
    if str1 != str2
      raise RuntimeError, "Given strings are not equal"
    end
  end

end

if __FILE__ == $0
  require "robotremoteserver"
  RobotRemoteServer.new(ExampleRemoteLibrary.new, *ARGV)
end

File: tools/remoteserver/example/examplelibrary.py
#!/usr/bin/env python

import os
import sys


class ExampleRemoteLibrary:

    def count_items_in_directory(self, path):
        return len([i for i in os.listdir(path) if not i.startswith('.')])

    def strings_should_be_equal(self, str1, str2):
        print "Comparing '%s' to '%s'" % (str1, str2)
        if str1 != str2:
            raise AssertionError("Given strings are not equal")


if __name__ == '__main__':
    from robotremoteserver import RobotRemoteServer
    RobotRemoteServer(ExampleRemoteLibrary(), *sys.argv[1:])
